src/main/java/org/folio/consortia/service/FolioTenantService.java

```diff
@@ -81,6 +94,50 @@ protected boolean tenantExists() {
     );
   }
 
+  /**
+   * In second time installing of module, method looks for system user permission changes
+   * if there is new changes in system user of central tenant,
+   * permission of shadow users will be updated with same ones with central system user in requesting tenant.
+   * This functionality is required by sharing settings because operation from central tenant context use system users,
+   * so each shadow users permissions also should be updated
+   */
+  private void updateLocalTenantShadowSystemUsers() {
+    if (!consortiaConfigurationService.isCentralTenantConfigurationExists()) {
+      log.info("The first time module is being installed, so skipping phase of updating shadow user permission");
+      return;
+    }
+
+    log.debug("updateLocalTenantShadowSystemUsers:: Trying to update shadow user permissions");
+    String requestingTenant = folioExecutionContext.getTenantId();
+    String centralTenantId = consortiaConfigurationService.getCentralTenantId(requestingTenant);
+
+    if (Objects.equals(requestingTenant, centralTenantId)) {
+      log.info("updateLocalTenantShadowSystemUsers:: consortia module is being installed on central tenant, no action required");
+      return;
+    }
+
+    // Fetch system user and its permission lists from central tenant
+    User centralSystemUser;
+    List<String> systemUserPermissionList;
+    try (var ignored = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(centralTenantId))) {
+      centralSystemUser = userService.getByUsername(systemUserUsername)
+        .orElseThrow(() -> new ResourceNotFoundException("systemUserUsername", systemUserUsername));
+      systemUserPermissionList = permissionsClient.getUserPermissions(centralSystemUser.getId()).getResult();
+    }
+
+    try (var ignored = new FolioExecutionContextSetter(contextHelper.getSystemUserFolioExecutionContext(requestingTenant))) {
```

Comment:
Now using context of requesting tenant to fix this error
![Untitled](https://github.com/folio-org/mod-consortia/assets/113523904/e7c56c83-326a-4fe5-8e56-a919d4b4a4c1)